File: my-folder/0020-valid-parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        openToClost = {"(": ")", "[": "]", "{": "}"}
        stack = []
        for c in s:
            if c in openToClost:
                stack.append(openToClost[c])
            else:
                if stack and stack[-1] == c:
                    stack.pop()
                else:
                    return False
        return len(stack) == 0

File: my-folder/0020-valid-parentheses/solution.ts
function isValid(s: string): boolean {
  const closeToOpen: { [key: string]: string } = {
    "(": ")",
    "[": "]",
    "{": "}",
  };
  const stack: string[] = [];

  for (const c of s) {
    if (c in closeToOpen) {
      stack.push(closeToOpen[c]);
    } else {
      if (stack.length > 0 && stack[stack.length - 1] === c) {
        stack.pop();
      } else {
        return false;
      }
    }
  }
  return stack.length === 0;
}
